pyiceberg/table/__init__.py

```diff
@@ -1551,6 +1627,17 @@ def append(self, df: pa.Table, snapshot_properties: Dict[str, str] = EMPTY_DICT)
         with self.transaction() as tx:
             tx.append(df=df, snapshot_properties=snapshot_properties)
 
+    def dynamic_overwrite(self, df: pa.Table, snapshot_properties: Dict[str, str] = EMPTY_DICT) -> None:
```

Comment:
I think there is consensus of moving everything to the transaction, instead of the table itself.

Reply:
yes the implementation of dynamic overwrite is inside transaction